src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests2.cs

```diff
@@ -2596,5 +2596,197 @@ public void M(string s)
+            0 => new E(), // 9.1
+            1 => new F(), // 9.2
+            _ => throw null,
+        });
+        _ = (C)(i switch // 10
+        {
+            0 => new A(),
+            1 => new B(),
+            2 => new C(),
+            _ => throw null,
+        });
+        _ = (D)(i switch // 11
+        {
+            0 => new A(),
+            1 => new B(),
+            2 => new C(),
+            _ => throw null,
+        });
+        _ = (D)(i switch // 12
+        {
+            0 => 1,
+            1 => 2,
+            _ => throw null,
+        });
+    }
+}
+
+class A : C { }
+class B : C { }
+class D
+{
+    public static implicit operator D(C c) => throw null;
+    public static implicit operator D(short s) => throw null;
+}
+class E
+{
+    public static implicit operator C(E c) => throw null;
+}
+class F
+{
+    public static implicit operator C(F c) => throw null;
+}
+";
+            var comp = CreateCompilation(source, parseOptions: TestOptions.RegularWithPatternCombinators).VerifyDiagnostics(
+                // (11,15): error CS8506: No best type was found for the switch expression.
+                //         _ = i switch // 1
+                Diagnostic(ErrorCode.ERR_SwitchExpressionNoBestType, "switch").WithLocation(11, 15),
+                // (25,18): error CS0029: Cannot implicitly convert type 'E' to 'D'
+                //             0 => new E(), // 3.1
+                Diagnostic(ErrorCode.ERR_NoImplicitConv, "new E()").WithArguments("E", "D").WithLocation(25, 18),
+                // (26,18): error CS0029: Cannot implicitly convert type 'F' to 'D'
+                //             1 => new F(), // 3.2
+                Diagnostic(ErrorCode.ERR_NoImplicitConv, "new F()").WithArguments("F", "D").WithLocation(26, 18),
+                // (63,18): error CS0029: Cannot implicitly convert type 'E' to 'D'
+                //             0 => new E(), // 9.1
+                Diagnostic(ErrorCode.ERR_NoImplicitConv, "new E()").WithArguments("E", "D").WithLocation(63, 18),
+                // (64,18): error CS0029: Cannot implicitly convert type 'F' to 'D'
+                //             1 => new F(), // 9.2
+                Diagnostic(ErrorCode.ERR_NoImplicitConv, "new F()").WithArguments("F", "D").WithLocation(64, 18)
+                );
+            var tree = comp.SyntaxTrees[0];
+            var model = comp.GetSemanticModel(tree);
+            var switches = tree.GetRoot().DescendantNodes().OfType<SwitchExpressionSyntax>().ToArray();
+            for (int i = 0; i < switches.Length; i++)
+            {
+                var @switch = switches[i];
+                var typeInfo = model.GetTypeInfo(@switch);
+                var conversion = model.GetConversion(@switch);
+                switch (i)
+                {
+                    case 0:
+                    case 7:
+                        Assert.Null(typeInfo.Type);
+                        Assert.Equal("C", typeInfo.ConvertedType.ToTestDisplayString());
```

Comment:
>Assert.Equal("C", typeInfo.ConvertedType.ToTestDisplayString()); [](start = 24, length = 64)

This looks unexpected for the explicit cast case. The ```ConvertedType``` is supposed to reflect an implicit conversion information rather than explicit conversion information. For example,  a ```null``` literal converted to something explicitly doesn't have either type.

Reply:
An explicit conversion is preferred by a cast over an implicit switch conversion.

---
In reply to: [440448568](https://github.com/dotnet/roslyn/pull/45069#discussion_r440448568) [](ancestors = 440448568)

Reply:
> An explicit conversion is preferred by a cast over an implicit switch conversion.

This has no impact on SemanticModel behavior. ```ConvertedType``` is not designed to reflect conversions for explicit cast in code.

---
In reply to: [440451109](https://github.com/dotnet/roslyn/pull/45069#discussion_r440451109) [](ancestors = 440451109,440448568)

Reply:
The switch expression conversion is never performed by the cast itself or any other enclosing node. It can only offer a target type. That is different than other target-typed conversions.

---
In reply to: [440452336](https://github.com/dotnet/roslyn/pull/45069#discussion_r440452336) [](ancestors = 440452336,440451109,440448568)